scan: remove CRLF from channel and provider name

athoik opened the pull request:
It seems some channels on 5E include CRLF on channel name.

[eDVBScan] SID 27ce: is scrambled!
[eDVBScan]   name 'Viasat History HD
', provider_name 'AMB'
[eDVBScan]   descr<5d>

Resulting in lamedb corruption, since we expect all data in one line.

This commit replaces CRLF with an empty string to fix that issue.

athoik committed:
scan: remove CRLF from channel and provider name

It seems some channels on 5E include CRLF on channel name.

[eDVBScan] SID 27ce: is scrambled!
[eDVBScan]   name 'Viasat History HD
', provider_name 'AMB'
[eDVBScan]   descr<5d>

Resulting in lamedb corruption, since we expect all data in one line.

This commit replaces CRLF with an empty string to fix that issue.

WanWizard commented:
My 2cts:

I would prefer a general solution, and replace "\r\n", "\r" and "\n" (in this order) by a space, and then trim the result. So "A\r\nB" becomes "A B" and not "AB".:

littlesat commented:
The provider makes here the mistake by copy and paste stuff with a Windows pc as the /r/n is at the end... we try to work-a-round here... then better work a round to by just strip the string (remove white spaces and other stuff) no spaces or \n \r \t at front and at the end.... and call it clean service name ‘patch’

athoik commented:
The most generic approach I can come with is the following: 

```
#include <iostream>
#include <string>
#include <regex>

std::string cleanNonGraph(std::string s)
{
        s = std::regex_replace(s, std::regex("[[^:graph:]]"), " ");
        s = std::regex_replace(s, std::regex("\\s{2,}"), " ");
        s = std::regex_replace(s, std::regex("^\\s+|\\s+$"), "");
        return s;
}

int main()
{
        std::string tmp("   |  test    a     test\r\r\t\t\r\n\n\n  a teare |      Ελληνικά;        <  <  < ");

        std::cout << tmp << std::endl;

        tmp = cleanNonGraph(tmp);

        std::cout << tmp << std::endl;

        return 0;

}
```

```
# g++ test_re.cpp && ./a.out
   |  test    a     test


  a teare |      Ελληνικά;        <  <  <
| test a test a teare | Ελληνικά; < < <
```


I can push function in estring class. I am not sure about the name :-)

littlesat commented:
I would call it strip.... or clean or so...

athoik committed:
estring: add new function strip_non_graph

Function strip_non_graph replaces all non visible characters with space,
then replaces double spaces with a signle space and finally strips whitespaces
from both ends.

athoik committed:
scan: use strip_non_graph on channel and provider name

This commit uses new function strip_non_graph on channel name
and provider name in order to ensure that all non graphical
characters removed. Also it strips whitespaces from both ends.

athoik committed:
estring: remove const reference from strip_non_graph
